api/client/lib.py

```diff
@@ -598,14 +598,19 @@ def get_geo_centre(access_token, api_host, region_id):
 
 
 @memoize(maxsize=None)
-def get_geojson(access_token, api_host, region_id):
-    url = '/'.join(['https:', '', api_host, 'v2/geocentres?includeGeojson=True&regionIds=' +
-                    str(region_id)])
+def get_geojsons(access_token, api_host, region_id, descendant_region_level_id=None):
```

Comment:
I was confused at first by this name because I thought it implied region_id could be a list.

I see now it's still an integer but you also specify region_level to get multiple results.

The endpoint does support lists of regionIds as inputs though, so if you wanted to do `get_geojsons([region['id'] for region in get_descendant_regions()])` you could do that as well, if the function supported it.

But right now that breaks on the `memoize` decorator, since you can't memoize lists as an input

Reply:
prefer to avoid ambiguous types in fn args, "int or list"  is an anti-pattern (ok for short term compatibility/deprecation/hacks :).  In general better to give the user the choice explicitly, singular or plural and only have the choice when necessary.  For this there is no current use case for mutliple region trees, i think it's efficient enough to do one tree at a time, since the root could be the whole world or a whole continent etc..  o

Reply:
Since it's a new function, I was thinking it could just require a list as an input, in contrast to the function we already have that takes an int.

If you only want one region, you could always just do `[region_id]`, but if we want to allow lists in the future now it'd have to be a 3rd separate "root" function that these two call. Getting deeper

Reply:
right but don't think there's a good use case, so odn't want to encourage unnecessarily large calls.   The real use case is process a region tree, there's n advantage to processing multiple trees in one batch, sinc ethe tree can be th ewhole world or a continent.